### PR TITLE
[OMNIML-2914] Fix export of fused layernorm weights for TE spec

### PR DESCRIPTION
**Pull request by yueshen2016**
## What does this PR do?

**Type of change:** Bug fix

**Overview:** When using TE spec (Transformer Engine layer specification), the `input_layernorm` and `pre_mlp_layernorm` are fused into `TELayerNormColumnParallelLinear`, making the standalone layernorm modules `IdentityOp`. The HF export (`export_mcore_gpt_to_hf`) skips `IdentityOp` modules, which silently dropped these layernorm weights from the exported HF checkpoint — producing an incomplete checkpoint missing `input_layernorm.weight` and `post_attention_layernorm.weight` for every layer.

This PR adds `_FusedLayerNormProxy`, a lightweight `nn.Module` that extracts the fused `layer_norm_weight` (and optional `layer_norm_bias`) from the TE linear module and exposes it as a standard `.weight` attribute. This allows the existing export rules to produce the correct HF key names without any changes to the rule mappings. The fix is applied to both transformer layer and Mamba layer export paths.

## Usage

No API changes. The fix is transparent — `export_mcore_gpt_to_hf` now correctly exports layernorm weights regardless of whether the model uses TE spec or local spec.

```python
# Same usage as before — no changes needed
import modelopt.torch.export as mtex
mtex.export_mcore_gpt_to_hf(model, hf_model_id, export_dir="./hf_export")
```

## Testing

- Quantized Llama-3.2-1B with TE spec via Megatron-Bridge `quantize.py`
- Exported to HF format via `export.py`
- Verified the exported safetensors contain `input_layernorm.weight` and `post_attention_layernorm.weight` for all layers
- Ran end-to-end inference with vLLM (`vllm/vllm-openai:latest`, `quantization="modelopt"`) — model loaded successfully and produced coherent text

## Before your PR is "*Ready for review*"

- **Make sure you read and follow [Contributor guidelines](https://github.com/NVIDIA/Model-Optimizer/blob/main/CONTRIBUTING.md)** and your commits are signed.
- **Is this change backward compatible?**: Yes — local spec models are unaffected (the `elif` branches only activate when layernorm is `IdentityOp` and fused weights exist)
- **Did you write any new necessary tests?**: No
- **Did you add or update any necessary documentation?**: No
- **Did you update [Changelog](https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?**: No

## Additional Information

The root cause is that TE spec fuses layernorm into the subsequent linear layer (`TELayerNormColumnParallelLinear`), storing the norm weight as `.layer_norm_weight` on the linear module instead of as `.weight` on a standalone norm module. The export walker checks `isinstance(layer.input_layernorm, IdentityOp)` and skips it, but never looks for the fused weight on the linear module. This fix bridges that gap with a zero-overhead proxy.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved export support for transformer and MTP models with fused layer normalization components, ensuring proper tensor mapping and export compatibility.

### CODE REVIEW

File: modelopt/torch/export/unified_export_megatron.py [hunk 1]
```diff
@@ -79,12 +79,97 @@
 
     has_mcore = True
 
+Qwen3VLModel = None
+try:
```
Comment (jenchen13) [inline modelopt/torch/export/unified_export_megatron.py:83]: does Qwen3VLModel exist in mcore? the other imports were from mcore. just wondering if this will cause a circular dependency since MBridge also depends on modelopt

File: modelopt/torch/export/unified_export_megatron.py [hunk 2]
```diff
@@ -542,22 +650,44 @@ def _get_transformer_layer_state_dict(self, layer, layer_id):
...
-                            expert.linear_fc1, layer_id, expert_id
+                            layer.mlp.experts.local_experts, layer_id
                         )
                         self.rules["local_experts.linear_fc2"](
-                            expert.linear_fc2, layer_id, expert_id
+                            layer.mlp.experts.local_experts, layer_id
                         )
                 else:
-                    # For llama 4, in hf unified checkpoint, all local experts share one scale
-                    self.rules["local_experts.linear_fc1"](
```
Comment (jenchen13) [inline modelopt/torch/export/unified_export_megatron.py:555]: does non-grouped MLP export still work even if you remove these lines?
